hnn_core/mpi_child.py

```diff
@@ -54,20 +54,20 @@ def __exit__(self, type, value, traceback):
             from mpi4py import MPI
             MPI.Finalize()
 
-    def _read_params(self):
-        """Read params broadcasted to all ranks on stdin"""
+    def _read_net(self):
```

Comment:
Nice catch! I hadn't thought about the `cell_responses`. Drive `'events'` get reset every time `simulate_dipole` is called. Should the same apply to `cell_responses`?

Fixed in 75c75bd. I don't think anything else gets appended?

Reply:
What on Earth is (was) [this doing](https://github.com/cjayb/hnn-core/blob/bab3ff3c2eafba2b2d5ef7cab8483c982be4432a/hnn_core/parallel_backends.py#L55), anyone?

```python
        net.gid_ranges = spikedata[2]  # only have one gid_ranges
```

It was introduced in #79. I'm pretty sure it's not doing anything, @jasmainak ?

I have a preference to _avoiding_ modifying/appending a `Network` during simulation, but I'm very much against  aggressively changing the `gid_ranges`!

Reply:
OK, removed in:

@ntolley why do you need this update? In other words: in what scenario do `net.gid_ranges` change during simulation?

```python
        net.cell_response.update_types(net.gid_ranges)
```

Reply:
Admittedly I've just carried this convention from older code, but I understand its purpose. It is necessary to create the `spike_types` variable which is an array of strings that pairs with `spike_times` and `spike_gids`. Since `spike_types` isn't explicitly passed from `network_builder`, it has to be created by referencing the `gid_ranges`. These variables are actually quite convenient for the plotting and analysis functions.  

Another reason this function exists is to maintain compatibility with HNN-GUI .spk files. Legacy files do not contain a `spike_types` column, the fix involves passing `gid_ranges` to `read_spikes()` [here](https://github.com/jonescompneurolab/hnn-core/blob/93c79d9f77a5ff7ad00fc00b6ef4a30a2ca9ccd5/hnn_core/network.py#L59).

Reply:
Is there any reason why we can't create a `CellResponse.gid_ranges` attribute that gets created in the constructor and updated in `CellResponse.read_spikes()` as well as at `simulate_dipole()` so that `CellResponse.gid_ranges` reflects all changes to any drive gids that get added prior to running the simulation? We can also save this for a later PR....

Reply:
> In other words: in what scenario do net.gid_ranges change during simulation?

But it doesn't. I guess this tells me the name of the method is not good. It should be called `update_spike_types` and you could pass `gid_ranges.copy()` to be even more explicit that `gid_ranges` does not change in the method.

I would really recommend keeping the discussion focused on the drives API, the PR is quite big as it is.

Reply:
@rythorpe unfortunately that won't work well since #217 will turn `CellResponse` into a list. 

Since that PR has the update methods, along with the `spikes_` attributes are living in the `Network` class, I think @jasmainak's makes sense.

Reply:
Resolving here due to irrelevance to present PR

Reply:
@cjayb We should probably undo the change in aaeba25 before preceding.

Reply:
How come?

Reply:
Oh I see. I just misunderstood the conversation. When I read this I thought that our conclusion was to either make it clear that `gid_ranges` is a copy or just save all changes for a later PR but I guess the update to `net.gid_ranges` needs to be removed regardless.

Reply:
Mhm, I guess I can see an argument for re-inserting that one line and dealing with it separately. However, removing it here will cause a rebase issue for said later PR, so it won't be forgotten ;)